README.md
epanet2
=======

EPANET, adjusted for Linux builds.

Please also have a look at epanetl, a localized version of EPANET:

http://ghydraulic.git.sourceforge.net/git/gitweb.cgi?p=ghydraulic/epanet;a=summary


ChangeLog
2010-02-23  Steffen Macke <sdteffen@sdteffen.de>

	* debian.changelog:
	* debian.control:
	* debian.rules:
	* epanet2.dsc: Fixed Debian/Ubuntu build

2009-05-19  Steffen Macke <sdteffen@sdteffen.de>

	* src/funcs.h:
	* src/input2.c:
	* src/toolkit.h: Reverting some changes (global variables problems)

2009-05-18  Steffen Macke <sdteffen@sdteffen.de>

	* src/epanet2.def:
	* src/funcs.h:
	* src/inpfile.c:
	* src/input2.c:
	* src/input3.c:
	* src/rules.c:
	* src/toolkit.h: Getting the DLL exports right for inp2shp
	* src/epanet2dll.dsp:
	* src/epanet2dll.dsw: MSVC project files to build epanet2.dll

2009-03-19  Steffen Macke <sdteffen@sdteffen.de>

	* src/epanet2.dev: Dev-C++ project to build epanet2.dll on Windows
	* src/epanet.c:
	* src/hydraul.c:
	* src/input1.c:
	* src/input2.c:
	* src/Makefile.am:
	* src/report.c:
	* src/rules.c: Small changes to make things work with Dev-C++ and without
	libintl
	* src/epanet2exe.dev: Dev-C++ project to build epanet2.exe on Windows

2008-09-12  Steffen Macke <sdteffen@sdteffen.de>

	* configure.in: Added AC_CANONICAL_TARGET
	* src/epanet.c: On Windows, use locale dir ..\locale relative to
	exe

2008-09-10  Steffen Macke <sdteffen@sdteffen.de>

	* INSTALL: Special Windows compilation instructions

2008-08-08  Steffen Macke <sdteffen@sdteffen.de>

	* src/epanet2.h: Additional #defines for inp2shp

2008-08-07  Steffen Macke <sdteffen@sdteffen.de>

	* doc/en/en2manual.pdf: Added EPANET manual

2008-07-24  Steffen Macke <sdteffen@sdteffen.de>

	* configure.in: Removed intl subfolder
	* src/report.c: Added some gettext calls
	* po/de.po: Some more German translations

2008-07-23  Steffen Macke <sdteffen@sdteffen.de>

	* po/de.po: Some more German translations

2008-07-22  Steffen Macke <sdteffen@sdteffen.de>

	* po/de.po: Some more German translations
	* debian.changelog:
	* debian.control:
	* debian.rules:
	* epanet2.dsc: Starting with Ubuntu build in openSUSE farm
	* epanet2.spec: Fist successful build in the openSUSE farm

2008-07-21  Steffen Macke <sdteffen@sdteffen.de>

	* po/de.po: Started work on the German translations
	* epanet2.spec: Various updates
	* po/Makefile.in:
	* po/POTFILES:
	* po/POTFILES.in:
	* po/de.po:
	* po/Makefile: Make "make dist" run again

2008-07-20  Steffen Macke <sdteffen@sdteffen.de>

	* src/i18n.h:
	* src/Makefile.in:
	* src/hydraul.c:
	* src/report.c:
	* src/epanet.c:
	* src/input1.c:
	* src/input2.c:
	* src/Makefile.am:
	* src/text.h:
	* src/rules.c:
	* po/POTFILES.in:
	* po/de.po:
	* po/epanet2.po: More work on the translations

2008-07-19  Steffen Macke <sdteffen@sdteffen.de>

	* Makefile.in:
	* configure.in:
	* src/Makefile.in:
	* src/epanet.c:
	* src/Makefile.am:
	* src/text.h:
	* po/Makevars:
	* po/de.po:
	* po/POTFILES.in:
	* po/POTFILES:
	* po/LINGUAS:
	* po/Makefile: Working on gettext() support
	* Makefile.am (SUBDIRS): Add po.
	(ACLOCAL_AMFLAGS): New variable.
	(EXTRA_DIST): Add config.rpath, m4/ChangeLog.
	* configure.in (AC_OUTPUT): Add po/Makefile.in.
	* AUTHORS: Mentioning Lewis Rossman
	* src/epanet.c:
	* src/Makefile.am: Also build shared library

2008-07-18  Steffen Macke <sdteffen@sdteffen.de>

	* Makefile.am:
	* Makefile.am:
	* epanet2.spec: Trying to fix openSUSE build
	* configure.in:
	* src/epanet.c:
	* src/Makefile.am:
	* epanet2.kdevelop: First Automake-based build
	* src/epanet.c:
	* src/toolkit.h: ENgetpumptype(), ENgetheadcurve() added
	* *.* The EPANET 2.00.12 release files from the EPA website


src/toolkit.h
/*
*******************************************************************

TOOLKIT.H - Prototypes for EPANET Functions Exported to DLL Toolkit

VERSION:    2.00
DATE:       5/8/00
            10/25/00
            3/1/01
            8/15/07    (2.00.11)
            2/14/08    (2.00.12)
AUTHOR:     L. Rossman
            US EPA - NRMRL

*******************************************************************
*/

// --- Define DLLEXPORT

#if defined(_MSC_VER)
#include <windows.h>
#ifdef DLL
#define DLLEXPORT extern __declspec(dllexport)
#else
#define DLLEXPORT extern __declspec(dllimport)
#endif
#else 
#define DLLEXPORT
#endif 


// --- Define the EPANET toolkit constants

#define EN_ELEVATION    0    /* Node parameters */
#define EN_BASEDEMAND   1
#define EN_PATTERN      2
#define EN_EMITTER      3
#define EN_INITQUAL     4
#define EN_SOURCEQUAL   5
#define EN_SOURCEPAT    6
#define EN_SOURCETYPE   7
#define EN_TANKLEVEL    8
#define EN_DEMAND       9
#define EN_HEAD         10
#define EN_PRESSURE     11
#define EN_QUALITY      12
#define EN_SOURCEMASS   13
#define EN_INITVOLUME   14
#define EN_MIXMODEL     15
#define EN_MIXZONEVOL   16

#define EN_TANKDIAM     17
#define EN_MINVOLUME    18
#define EN_VOLCURVE     19
#define EN_MINLEVEL     20
#define EN_MAXLEVEL     21
#define EN_MIXFRACTION  22
#define EN_TANK_KBULK   23

#define EN_DIAMETER     0    /* Link parameters */
#define EN_LENGTH       1
#define EN_ROUGHNESS    2
#define EN_MINORLOSS    3
#define EN_INITSTATUS   4
#define EN_INITSETTING  5
#define EN_KBULK        6
#define EN_KWALL        7
#define EN_FLOW         8
#define EN_VELOCITY     9
#define EN_HEADLOSS     10
#define EN_STATUS       11
#define EN_SETTING      12
#define EN_ENERGY       13

#define EN_DURATION     0    /* Time parameters */
#define EN_HYDSTEP      1
#define EN_QUALSTEP     2
#define EN_PATTERNSTEP  3
#define EN_PATTERNSTART 4
#define EN_REPORTSTEP   5
#define EN_REPORTSTART  6
#define EN_RULESTEP     7
#define EN_STATISTIC    8
#define EN_PERIODS      9

#define EN_NODECOUNT    0   /* Component counts */
#define EN_TANKCOUNT    1
#define EN_LINKCOUNT    2
#define EN_PATCOUNT     3
#define EN_CURVECOUNT   4
#define EN_CONTROLCOUNT 5

#define EN_JUNCTION     0    /* Node types */
#define EN_RESERVOIR    1
#define EN_TANK         2

#define EN_CVPIPE       0    /* Link types. */
#define EN_PIPE         1    /* See LinkType in TYPES.H */
#define EN_PUMP         2
#define EN_PRV          3
#define EN_PSV          4
#define EN_PBV          5
#define EN_FCV          6
#define EN_TCV          7
#define EN_GPV          8

#define EN_NONE         0    /* Quality analysis types. */
#define EN_CHEM         1    /* See QualType in TYPES.H */
#define EN_AGE          2
#define EN_TRACE        3

#define EN_CONCEN       0    /* Source quality types.      */
#define EN_MASS         1    /* See SourceType in TYPES.H. */
#define EN_SETPOINT     2
#define EN_FLOWPACED    3

#define EN_CFS          0    /* Flow units types.   */
#define EN_GPM          1    /* See FlowUnitsType   */
#define EN_MGD          2    /* in TYPES.H.         */
#define EN_IMGD         3
#define EN_AFD          4
#define EN_LPS          5
#define EN_LPM          6
#define EN_MLD          7
#define EN_CMH          8
#define EN_CMD          9

#define EN_TRIALS       0   /* Misc. options */
#define EN_ACCURACY     1
#define EN_TOLERANCE    2
#define EN_EMITEXPON    3
#define EN_DEMANDMULT   4

#define EN_LOWLEVEL     0   /* Control types.  */
#define EN_HILEVEL      1   /* See ControlType */
#define EN_TIMER        2   /* in TYPES.H.     */
#define EN_TIMEOFDAY    3

#define EN_AVERAGE      1   /* Time statistic types.    */
#define EN_MINIMUM      2   /* See TstatType in TYPES.H */
#define EN_MAXIMUM      3
#define EN_RANGE        4

#define EN_MIX1         0   /* Tank mixing models */
#define EN_MIX2         1
#define EN_FIFO         2
#define EN_LIFO         3

#define EN_NOSAVE       0   /* Save-results-to-file flag */
#define EN_SAVE         1

#define EN_INITFLOW    10   /* Re-initialize flows flag  */

/*** Updated 11/24/06 ***/
#define EN_CONST_HP     0   /*    constant horsepower              */
#define EN_POWER_FUNC   1   /*    power function                   */
#define EN_CUSTOM       2   /*    user-defined custom curve        */


// --- Declare the EPANET toolkit functions

 int  DLLEXPORT ENepanet(char *, char *, char *, void (*) (char *));

 int  DLLEXPORT ENopen(char *, char *, char *);
 int  DLLEXPORT ENsaveinpfile(char *);
 int  DLLEXPORT ENclose(void);

 int  DLLEXPORT ENsolveH(void);
 int  DLLEXPORT ENsaveH(void);
 int  DLLEXPORT ENopenH(void);
 int  DLLEXPORT ENinitH(int);
 int  DLLEXPORT ENrunH(long *);
 int  DLLEXPORT ENnextH(long *);
 int  DLLEXPORT ENcloseH(void);
 int  DLLEXPORT ENsavehydfile(char *);
 int  DLLEXPORT ENusehydfile(char *);

 int  DLLEXPORT ENsolveQ(void);
 int  DLLEXPORT ENopenQ(void);
 int  DLLEXPORT ENinitQ(int);
 int  DLLEXPORT ENrunQ(long *);
 int  DLLEXPORT ENnextQ(long *);
 int  DLLEXPORT ENstepQ(long *);
 int  DLLEXPORT ENcloseQ(void);

 int  DLLEXPORT ENwriteline(char *);
 int  DLLEXPORT ENreport(void);
 int  DLLEXPORT ENresetreport(void);
 int  DLLEXPORT ENsetreport(char *);

 int  DLLEXPORT ENgetcontrol(int, int *, int *, float *,
                int *, float *);
 int  DLLEXPORT ENgetcount(int, int *);
 int  DLLEXPORT ENgetoption(int, float *);
 int  DLLEXPORT ENgettimeparam(int, long *);
 int  DLLEXPORT ENgetflowunits(int *);
 int  DLLEXPORT ENgetpatternindex(char *, int *);
 int  DLLEXPORT ENgetpatternid(int, char *);
 int  DLLEXPORT ENgetpatternlen(int, int *);
 int  DLLEXPORT ENgetpatternvalue(int, int, float *);
 int  DLLEXPORT ENgetqualtype(int *, int *);
 int  DLLEXPORT ENgeterror(int, char *, int);

 int  DLLEXPORT ENgetnodeindex(char *, int *);
 int  DLLEXPORT ENgetnodeid(int, char *);
 int  DLLEXPORT ENgetnodetype(int, int *);
 int  DLLEXPORT ENgetnodevalue(int, int, float *);

 int  DLLEXPORT ENgetlinkindex(char *, int *);
 int  DLLEXPORT ENgetlinkid(int, char *);
 int  DLLEXPORT ENgetlinktype(int, int *);
 int  DLLEXPORT ENgetlinknodes(int, int *, int *);
 int  DLLEXPORT ENgetlinkvalue(int, int, float *);

 int  DLLEXPORT ENgetversion(int *);

 int  DLLEXPORT ENsetcontrol(int, int, int, float, int, float);
 int  DLLEXPORT ENsetnodevalue(int, int, float);
 int  DLLEXPORT ENsetlinkvalue(int, int, float);
 int  DLLEXPORT ENaddpattern(char *);
 int  DLLEXPORT ENsetpattern(int, float *, int);
 int  DLLEXPORT ENsetpatternvalue(int, int, float);
 int  DLLEXPORT ENsettimeparam(int, long);
 int  DLLEXPORT ENsetoption(int, float);
 int  DLLEXPORT ENsetstatusreport(int);
 int  DLLEXPORT ENsetqualtype(int, char *, char *, char *);

/*** Updated 11/24/06 ***/
 int DLLEXPORT ENgetheadcurve(int, char *);
 int DLLEXPORT ENgetpumptype(int, int *);

